gradio/blocks.py

```diff
@@ -3033,3 +3065,30 @@ def get_event_targets(
             event = getattr(block, event_name)
             target_events.append(event)
         return target_events
+
+    @document()
+    def route(self, name: str, path: str | None = None):
+        """
+        Adds a new page to the Blocks app.
+        Parameters:
+            name: The name of the page as it appears in the nav bar.
+            path: The url path of the page. If not provided, it is generated from the name.
+        Example:
+            with gr.Blocks() as demo:
+                name = gr.Textbox(label="Name")
+                ...
+            with demo.route("Test", "/test"):
+                num = gr.Number()
+                ...
+        """
+        if path:
+            path = path.strip("/")
+        if path in EXISTING_ROUTES:
+            raise ValueError(f"Route with path '{path}' already exists")
+        if path is None:
+            path = name.lower().replace(" ", "-")
+        if path in EXISTING_ROUTES:
+            path = "_" + path
```

Comment:
Pick one: do you either raise an error or prepend a "_" to existing routes 😆

Reply:
if you explicitly set a route and it isnt allowed, there's an error. But if you just provide a link name, and the autogenerate link route for that name is invalid, we prepend an underscore.